endpoints/remotehosts/remotehosts.py

```diff
@@ -101,7 +101,7 @@ def remote_connection(host, user):
                     validate_comment(msg)
                 else:
                     log.info(msg)
-                break
```

Comment:
did this cause any side effects ?

Reply:
It was causing the code to iterate through the loop for at least 2 iterations, even if it succeeded on the first.  This would result in the `connection.open()` being called at least twice no matter what.  It's not exactly clear to me what calling that a second time after it had succeeded the first time might mean, but I could see that it might have added some extra overhead.

It was also weird that it was logging that it took a 2nd attempt when there was no logged failed attempt.  This is what clued me into there being a problem.